website/integrations: add OVH

hobyte opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
Add OVH Cloud integration guide

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

Commit:
Add documentation for OVH integration

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+
+## Service Configuration
+
+1. Go to [OVH Cloud](https://www.ovhcloud.com) and log in.
```
```suggestion
1. Log in to the [OVHcloud Control Panel](https://www.ovh.com/auth/).
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+
+3. Click **Submit** to save the new application and provider.
+
+## Service Configuration
```
```suggestion
## OVHcloud Configuration
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+
+## authentik Configuration
+
+To support the integration of OVH Cloud with authentik, you need to create an application/provider pair in authentik.
```
```suggestion
To support the integration of OVHcloud with authentik, you need to create an application/provider pair in authentik.
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+
+## What is OVH Cloud
+
+> OVH Cloud is a French cloud provider. They provide public and private cloud products, shared hosting, and dedicated servers in 140 countries.
```
```suggestion
> OVHcloud is a French cloud provider. They provide public and private cloud products, shared hosting, and dedicated servers in 140 countries.
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+---
+title: Integrate with OVH Cloud
```
```suggestion
title: Integrate with OVHcloud
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+---
+title: Integrate with OVH Cloud
+sidebar_label: OVH Cloud
```
```suggestion
sidebar_label: OVHcloud
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+support_level: community
+---
+
+## What is OVH Cloud
```
```suggestion
## What is OVHcloud
```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+    - **Configure Bindings** _(optional)_: You can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.
+
+3. Click **Submit** to save the new application and provider.
+
```
```suggestion

### Download metadata file

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click on the name of the provider that you created in the previous section (e.g. `Provider for OVHcloud`).
3. Under **Related objects** > **Metadata**, click on **Download**. This downloaded file is your `SAML Metadata` file and it will be required in the next section.

```

dewi-tik commented on website/integrations/services/ovh/index.md:
```diff
@@ -0,0 +1,53 @@
+4. Click the **Identities** tab to access local users management and switch to the **SSO** tab.
+5. Click on the **SSO Connection** button.
+    - If you want to keep local OVH users, tick the `Keep active OVHcloud users` box.
+    - Copy the XML-Metadata from your authentik installation (Applications -> Providers -> Your OVH Provider -> Metadata) to the **XML-Metadata** field.
```
This will need to be updated to explain in more detail how to paste in the metadata information.

hobyte replied:
I tried. Hopefully, this description is more detailed, but I cannot find a phrasing I'm really happy with.

dewi-tik commented on website/sidebars/integrations.mjs:
```diff
@@ -54,6 +54,7 @@ const items = [
             "services/google/index",
             "services/hashicorp-cloud/index",
             "services/oracle-cloud/index",
+            "services/ovh/index",
```
```suggestion
            "services/ovhcloud/index",
```

Could you also please update the file path of the `index.md` file to `integrations/services/ovhcloud/index.md`

dewi-tik reviewed (changes requested):
Thank you very much for the contribution @hobyte. It's very much appreciated.

I've made a few suggestions to start off the review.

Commit:
Apply suggested changes

Commit:
update file path to reflect service name change

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+## Preparation
+
+The following placeholders are used in this guide:
+
```
Please remove the extra newline

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+    - **Application**: Provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
+    - **Choose a Provider type**: Select **SAML Provider** as the provider type.
+    - **Configure the Provider**: Provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations:
+        - Set the **ACS URL** to `https://www.ovhcloud.com/eu/auth/saml/acs` for EU region, or `https://www.ovhcloud.com/ca/auth/saml/acs` for CA region.
```
If you're somewhere else than canada or the eu, what value is used?

dewi-tik replied:
They only seem to have EU and CA for these URLs. They only mention these two in their SSO docs: https://help.ovhcloud.com/csm/en-connect-saml-sso-okta?id=kb_article_view&sysparm_article=KB0057810

dominic-r replied:
Interesting (?)

dominic-r replied:
Welp IG everyone is either in the EU or CA

hobyte replied:
For the us, I found this url https://us.ovhcloud.com/auth/ in their docs (https://us.ovhcloud.com/auth/). As I cannot find a list or something with the URLs for all regions, I add all I know of right now and hope there aren't any more.

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+### Download metadata file
+
+1. Log in to authentik as an admin, and open the authentik Admin interface.
+2. Navigate to **Applications** > **Providers** and click on the name of the provider that you created in the previous section (e.g. `Provider for OVHcloud`).
```
```suggestion
2. Navigate to **Applications** > **Providers** and click on the name of the provider that you created in the previous section.
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+
+1. Log in to authentik as an admin, and open the authentik Admin interface.
+2. Navigate to **Applications** > **Providers** and click on the name of the provider that you created in the previous section (e.g. `Provider for OVHcloud`).
+3. Under **Related objects** > **Metadata**, click on **Download**. This downloaded file is your `SAML Metadata` file and it will be required in the next section.
```
```suggestion
3. Under **Related objects** > **Metadata**, click on **Download**. This downloaded file is your **SAML Metadata** file and it will be required in the next section.
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+
+## OVHcloud Configuration
+
+1. Log in to the [OVHcloud Control Panel](https://www.ovh.com/auth/).
```
Technically, this url wouldn't work for all as I login on ca.ovh.com and entering credentials on the url provided returns an error

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+3. Select **Identity and Access Management (IAM)** from the left-hand menu.
+4. Click the **Identities** tab to access local users management and switch to the **SSO** tab.
+5. Click on the **SSO Connection** button.
+    - If you want to keep local OVH users, tick the `Keep active OVHcloud users` box.
```
```suggestion
    - If you want to keep local OVH users, tick the **Keep active OVHcloud users** box.
```

since it's a value in UI

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+    - If you want to keep local OVH users, tick the `Keep active OVHcloud users` box.
+    - Open the File you downloaded in [Download metadata file](#download-metadata-file). Copy the content and paste it into the **XML-Metadata** field.
+6. Close the window by clicking **Confirm**.
+
```
Can we include a references section which points to official docs?

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+
+To verify that authentik is correctly integrated with OVH Cloud, first log out of your account. On the OVHcloud login page, enter your [OVH Customer ID/NIC handle](https://help.ovhcloud.com/csm/en-account-create-ovhcloud-account?id=kb_article_view&sysparm_article=KB0043022#what-is-my-nic-handle) followed by `/idp` (e.g. `xx1111-ovh/idp`) without entering a password, and click the Login button.
+
+You will be redirected to your authentik installation for authentication. Once authenticated you will be logged in to OVH Cloud.
```
```suggestion
You will be redirected to your authentik installation for authentication. Once authenticated you will be logged in to OVHcloud.
```

to match others

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,61 @@
+## OVHcloud Configuration
+
+1. Log in to the [OVHcloud Control Panel](https://www.ovh.com/auth/).
+2. Click on your **name** in the top right corner. Then, click on your **name** again in the sidebar that opens.
```
```suggestion
2. Click your name in the top right corner. In the sidebar that appears, click your name again.
```

dominic-r reviewed (changes requested):
Thanks for your contribution, a few more comments to make it perfect!

Commit:
Fix formatting

Commit:
Improve ACS URL options

- add option for US region
- add section with references to OVHcloud help center

hobyte committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: hobyte <63231858+hobyte@users.noreply.github.com>

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+## References
+
+- [User management & Federation - OVHcloud help center](https://help.ovhcloud.com/csm/en-ie-documentation-manage-operate-user-federation?id=kb_browse_cat&kb_id=3d4a8129a884a950f07829d7d5c75243&kb_category=21734cbe50d47d90476b12dfd60b3542&spa=1)
```
```suggestion
- [OVHcloud Help Center - User management & Federation](https://help.ovhcloud.com/csm/en-ie-documentation-manage-operate-user-federation?id=kb_browse_cat&kb_id=3d4a8129a884a950f07829d7d5c75243&kb_category=21734cbe50d47d90476b12dfd60b3542&spa=1)
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+## References
+
+- [User management & Federation - OVHcloud help center](https://help.ovhcloud.com/csm/en-ie-documentation-manage-operate-user-federation?id=kb_browse_cat&kb_id=3d4a8129a884a950f07829d7d5c75243&kb_category=21734cbe50d47d90476b12dfd60b3542&spa=1)
+- [User management & Federation - OVHcloud US help center](https://support.us.ovhcloud.com/hc/en-us/sections/27230986868883-Federation)
```
```suggestion
- [OVHcloud US Help Center - User management & Federation](https://support.us.ovhcloud.com/hc/en-us/sections/27230986868883-Federation)
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+## Configuration Verification
+
+To verify that authentik is correctly integrated with OVH Cloud, first log out of your account. On the OVHcloud login page, enter your [OVH Customer ID/NIC handle](https://help.ovhcloud.com/csm/en-account-create-ovhcloud-account?id=kb_article_view&sysparm_article=KB0043022#what-is-my-nic-handle) followed by `/idp` (e.g. `xx1111-ovh/idp`) without entering a password, and click the Login button.
```
```suggestion
To verify that authentik is properly integrated with OVHcloud, start by logging out of your account. On the OVHcloud login page, enter your [OVH Customer ID/NIC handle](https://help.ovhcloud.com/csm/en-account-create-ovhcloud-account?id=kb_article_view&sysparm_article=KB0043022#what-is-my-nic-handle) followed by `/idp` (e.g., `xx1111-ovh/idp`)—leave the password field blank—and click **Login**.
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+To verify that authentik is correctly integrated with OVH Cloud, first log out of your account. On the OVHcloud login page, enter your [OVH Customer ID/NIC handle](https://help.ovhcloud.com/csm/en-account-create-ovhcloud-account?id=kb_article_view&sysparm_article=KB0043022#what-is-my-nic-handle) followed by `/idp` (e.g. `xx1111-ovh/idp`) without entering a password, and click the Login button.
+
+You will be redirected to your authentik installation for authentication. Once authenticated you will be logged in to OVHcloud.
```
```suggestion
You’ll be redirected to your authentik instance to complete authentication. Once successful, you’ll be logged in to OVHcloud.
```

dominic-r commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+## OVHcloud Configuration
+
+1. Log in to the **OVHcloud Control Panel**.
```
```suggestion
1. Log in to the OVHcloud Control Panel.
```

dominic-r replied:
When you load into it:
![IMG_1082](https://github.com/user-attachments/assets/0b284604-db4d-4f43-8576-2a227fdffdce)

dominic-r reviewed (approved):
After that's done, LGTM

tanberry commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+### Create an Application and Provider in authentik
+
+1. Log in to authentik as an admin and open the authentik Admin interface.
```
```suggestion
1. Log in to authentik as an administrator and open the authentik Admin interface.
```

tanberry commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+
+### Download metadata file
+
+1. Log in to authentik as an admin, and open the authentik Admin interface.
```
```suggestion
1. Log in to authentik as an administrator and open the authentik Admin interface.
```

tanberry commented on website/integrations/services/ovhcloud/index.md:
```diff
@@ -0,0 +1,68 @@
+    - Open the File you downloaded in [Download metadata file](#download-metadata-file). Copy the content and paste it into the **XML-Metadata** field.
+6. Close the window by clicking **Confirm**.
+
+## Configuration Verification
```
```suggestion
## Configuration verification
```

tanberry reviewed (dismissed):
Thanks so much for this contribution, @hobyte ! A few edit requests, and then as soon as those are in let's get this merged.

Commit:
Apply suggestions

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik committed:
Update website/integrations/services/ovhcloud/index.md

Co-authored-by: Dominic R <dominic@sdko.org>
Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

dewi-tik reviewed (approved):
Thanks for the contribution @hobyte!

dewi-tik committed:
Minor changes

dewi-tik committed:
Merge remote-tracking branch 'upstream/main' into ovh-integration

dewi-tik reviewed (approved):
I made some minor changes to fix the prettier error. Thanks again @hobyte. We'll merge once we're able to get this to pass the checks.

dewi-tik committed:
prettier fix

Signed-off-by: Dewi Roberts <dewi@goauthentik.io>

tanberry reviewed (approved):
Thanks so much @hobyte !

dewi-tik committed:
Merge remote-tracking branch 'upstream/main' into ovh-integration
